### ISSUE TITLE
Rule Request: Redundant `final` for `actor`s

### ISSUE THREAD

**Issue by akuzminskyi** [realm/SwiftLint#6407]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New Rule Request

`actor`s in Swift already `final` by default. but developers still sometimes write it as `final actor` and it doesn't produce any compiler warnings. But basically `final` is redundant in this case. similar to `internal` ACL.

I don't see any configuration for the rule. and I think it should be enabled by default as it doesn't change "logic".

```
final actor DataStorage {
```
-> 
```
actor DataStorage {
```

https://github.com/swiftlang/swift-evolution/blob/main/proposals/0306-actors.md#actors-2
> hey are reference types like classes, but do not support inheritance, and therefore do not have (or need) features such as `required` and `convenience` initializers, `overriding`, or class members, `open` and `final`.
